File: Java/Easy/2053.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        // use a hashmap to store the frequency count of the strings
        HashMap<String, Integer> map = new HashMap<>();

        // fill the map with the frequencies
        for (String string : arr) {
            int temp = map.getOrDefault(string, 0);
            temp++;
            map.put(string, temp);
        }

        // find the k distinct string
        for (String string : arr) {
            int temp = map.get(string);
            if (temp == 1) {
                k--;
            }

            if (k == 0) {
                return string;
            }
        }

        return "";
    }
}

File: Python/Easy/2053.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        # use a dictionary to act as a frequency map
        freq = {}

        # fill the dictionary
        for string in arr:
            temp = freq.get(string, 0)
            temp += 1
            freq[string] = temp

        # check the kth item in the array by comparing to dictionary
        for string in arr:
            temp = freq.get(string)
            if temp == 1:
                k -= 1
            if k == 0:
                return string

        return ""
